Data/Text.hs

```diff
@@ -7,6 +7,10 @@
 {-# LANGUAGE TypeFamilies #-}
 #endif
 
+#if defined(__GLASGOW_HASKELL__) && !defined(__HADDOCK__)
```

Comment:
Have you tried generating docs with haddock? I am not sure about not including this for haddock. It runs CPP just as ghc. And I imagine it will fail if it can't find `WORD_SIZE_IN_BITS`.